srml/contract/src/lib.rs

```diff
@@ -342,11 +389,14 @@ decl_storage! {
 		pub PristineCode: map CodeHash<T> => Option<Vec<u8>>;
 		/// A mapping between an original code hash and instrumented wasm code, ready for the execution.
 		pub CodeStorage: map CodeHash<T> => Option<wasm::PrefabWasmModule>;
+		/// The subtrie counter associated with an account id
+		pub AccountCounter: u64 = 0;
 	}
 }
 
-/// The storage items associated with an account/key.
-///
+/// The storage items associated with a prefix space in a tree
+/// TODO can we remove it, seems usefull for inner implementation
+/// just not in the right place
```

Comment:
It is an almost working implementation (from Gav last remark I see that it really need at least a small fix on full_key calculation to have the key2 part made unique from key1).
So yes there is no reason, just a tool that may prove being useful for other runtime.

Reply:
I mean, why we can't delete it from here? It looks like this definition is not used anywhere and might be confusing.

I am so curious about this because we have a convention for using TODOs, basically we are trying to avoid them in Substrate and leave only those which are justified and have an issue in the issue tracker.

I'd say that this particular `StorageDoubleMap` instantiation is unused and thus useless (Note that `StorageDoubleMap` might be a useful primitive, and I'm not proposing to remove it).

Reply:
It is really not used, the TODO behing here to ask if we want to keep the functionnality, seems like we do not want to keep unused code, so let's remember f2bd95a9a53bd001d7fd988ecdfb486da88565e6 if we want to revert the deletion of this one.

Reply:
Works for me!